### PR TITLE
feat: add `dropzone` in upload page

### PR DESCRIPTION
**Pull request by tokitou-san**

### CODE REVIEW

File: src/routes/upload/+page.svelte [hunk 1]
```diff
@@ -46,6 +48,46 @@
...
+        show_dropzone = true;
+    }
+
+    function on_dragleave_hander(event: DragEvent) {
+        event.preventDefault();
+        show_dropzone = false;
+    }
+
+    function on_dropzone_dragover() {
+        const dropzone_element = document.querySelector("dropzone") as HTMLDivElement;
```
Comment (baseplate-admin) [inline src/routes/upload/+page.svelte:80]: Bad way. Use bindings

File: src/routes/upload/+page.svelte [hunk 2]
```diff
@@ -46,6 +48,46 @@
...
+        // change background
+        dropzone_element.classList.add("bg-surface-500/25");
+    }
+
+    function on_dropzone_dragleave() {
+        const dropzone_element = document.querySelector("dropzone") as HTMLDivElement;
```
Comment (baseplate-admin) [inline src/routes/upload/+page.svelte:86]: Bad way. Use bindings

File: src/routes/upload/+page.svelte [hunk 3]
```diff
@@ -60,6 +102,31 @@
     {@html opengraph_html}
 </svelte:head>
 
+<svelte:window
+    on:dragover={on_dragover_handler}
+    on:dragleave={on_dragleave_hander}
```
Comment (baseplate-admin) [inline src/routes/upload/+page.svelte:107]: use `preventDefault` pipe instead of explicitly preventing it.

**Review by baseplate-admin — CHANGES_REQUESTED** [review]

File: src/routes/upload/+page.svelte [hunk 4]
```diff
@@ -46,6 +49,25 @@
...
+        const file_list_names = data_list.map((data) => {
+            return data.file.name;
+        });
+
+        Array.from(files).forEach((file) => {
+            if (file.type === "video/mp4" || file.type === "video/mkv") {
+                if (!file_list_names.includes(file.name)) {
+                    data_list = data_list.concat({ file: file });
+                }
+            }
```
Comment (baseplate-admin) [inline src/routes/upload/+page.svelte:67]: bad logic

File: src/routes/upload/+page.svelte [hunk 5]
```diff
@@ -13,10 +13,13 @@
     import { ProgressBar } from "@skeletonlabs/skeleton";
     import dayjs from "dayjs";
     import prettyBytes from "pretty-bytes";
+    import { blur } from "svelte/transition";
 
     let main_checkbox: boolean | undefined;
     let checkbox_elements: Array<HTMLInputElement> = new Array<HTMLInputElement>();
     let data_list: Array<{ file: File }> = new Array<{ file: File }>();
+    let show_dropzone = false;
+    let dropzone_active = false;
```
Comment (baseplate-admin) [inline src/routes/upload/+page.svelte:22]: dont use 2 independent variable showing same thing

**Comment by baseplate-admin** [comment]
Folder Drag is not possible

File: src/routes/upload/+page.svelte [hunk 6]
```diff
@@ -74,18 +74,37 @@
...
-            if (Object.keys(file_whitelist).includes(file.type)) {
-                if (!file_list_names.includes(file.name)) {
-                    data_list = data_list.concat({ file: file });
-                }
+        Array.from(files).forEach(async (item) => {
+            const file = item.webkitGetAsEntry();
+            
+            if(file?.isDirectory){
+                const files = await scan_directory(file as FileSystemDirectoryEntry);
+                console.log("I cant see", files);
```
Comment (tokitou-san) [inline src/routes/upload/+page.svelte:87]: `files` shows as `undefined`
but it consoles in `scan_directory` function

Comment (baseplate-admin) [reply src/routes/upload/+page.svelte:87]: Oh well, welcome to callback hell

**Pull request merged**